lib/galaxy/jobs/command_factory.py

```diff
@@ -19,6 +19,13 @@
 SETUP_GALAXY_FOR_METADATA = """
 [ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
 """
+PREPARE_DIRS = """mkdir -p working outputs
+if [ -f .job_started ]; then
```

Comment:
Actually this is redundant, I could just do
```suggestion
if [ -d _working]; then
```

and skip the `touch` below.